index.src.html

```diff
@@ -501,7 +501,8 @@ <h3 id="dom-api">JavaScript API</h3>
       Clears data based on the values in the |options| argument.
       Returns a Promise that resolves when clearing is complete. If no
       {{StorageClearOptions/types}} are specified, all data types will be
-      cleared.
+      cleared. See [[#clear-api]] for the formal description of the clearing
+      algorithm.
```

Comment:
So... I read through a few other specs, but it's still difficult for me to tell how exactly it should be done, because of course, everyone does it differently.

Here's what I did:
1. Most algorithm definitions have a short preface text before enumerating the steps; where they didn't have it, I added it. In that preface I used dfn around the name of the algorithm, and I'm linking to that dfn instead of the section from the rest of the document. @annevk, I think this is actually the style you use :)

2. I checked the bikeshed documentation about "algorithm"; AFAIU it can't be used to annotate a dfn to say that it's an algorithm. It should be used with section or header to scope the algorithm's variables. So I added this annotation to every header. The value is always identical to the dfn, so in an indirect way, I did manage to semantically annotate the dfn's.